### PR TITLE
[Feature] FakeSource support MultipleTable

### PR DESCRIPTION
**Pull request by ruanwenjun**
### Purpose of this pull request

FakeSource support generate multiple different CatalogTable. This will be used for testing multiple table feature for sink.

### How was this patch tested?
Tested by UT and e2e case

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [X] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/config/MultipleTableFakeSourceConfig.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @Getter private List<FakeConfig> fakeConfigs;
+
+    public MultipleTableFakeSourceConfig(ReadonlyConfig fakeSourceRootConfig) {
+        if (fakeSourceRootConfig.getOptional(FakeOption.TABLES_CONFIGS).isPresent()) {
+            parseFromConfigs(fakeSourceRootConfig);
+        } else {
+            parseFromConfig(fakeSourceRootConfig);
+        }
+        // validate
+        assert fakeConfigs != null;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/config/MultipleTableFakeSourceConfig.java]: I think the `assert` will not work by default. Please use `com.google.common.base.Preconditions.checkNotNull`.

File: seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeDataTypeConvertor.java [hunk 2]
```diff
@@ -209,9 +209,9 @@ public SeaTunnelDataType<?> toSeaTunnelType(
             case DATE:
                 return LocalTimeType.LOCAL_DATE_TYPE;
             case DATETIME:
-                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
-            case TIMESTAMP:
                 return LocalTimeType.LOCAL_TIME_TYPE;
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeDataTypeConvertor.java:212]: mapping DATETIME to LOCAL_DATE_TIME_TYPE ?

File: seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/config/MultipleTableFakeSourceConfig.java (hunk 1 above)
Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/config/MultipleTableFakeSourceConfig.java]: Done

File: seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeDataTypeConvertor.java (hunk 2 above)
Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeDataTypeConvertor.java:212]: The DATETIME should convert to LocalTime and TIMESTAMP should convert to  LOCAL_DATE_TIME_TYPE

Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/catalog/MaxComputeDataTypeConvertor.java:212]: I revert this change.

**Review by Hisoka-X — COMMENTED** [review]
Please remove https://github.com/apache/seatunnel/pull/5766/files#diff-f457bd8e001241b29148166779a736845b3faa6e0259f3a402bc3fa58f75fb3eR104

**Comment by ruanwenjun** [comment]
> Please remove https://github.com/apache/seatunnel/pull/5766/files#diff-f457bd8e001241b29148166779a736845b3faa6e0259f3a402bc3fa58f75fb3eR104

Done

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**